epl-nft/contracts/EnglishPremierLeague.cdc

```diff
@@ -0,0 +1,854 @@
+    }
+
+    /// Set getters
+    ///
+    pub fun getSet(id: UInt64): EnglishPremierLeague.Set? {
+        return EnglishPremierLeague.setByID[id]
+    }
+
+    pub fun getSetByName(name: String): EnglishPremierLeague.Set? {
+        let id = EnglishPremierLeague.setIDByName[name]!
+
+        if id == nil {
+            return nil
+        }
+
+        return EnglishPremierLeague.setByID[id!]
+    }
+
+    pub fun getAllSetNames(): [String] {
+        return EnglishPremierLeague.setIDByName.keys
+    }
+
+    /// A top-level Tag with ID and name
+    /// Examples include goal, match winner, assist, header, save, brace
+    ///
+    pub struct Tag {
+        pub let id: UInt64
+        pub let name: String
+
+        init (id: UInt64, name: String) {
+            if let tag = EnglishPremierLeague.tagByID[id] {
+                self.id = tag.id
+                self.name = tag.name
+            } else {
+                self.id = id
+                self.name = name
+            }
+        }
+    }
+
+    /// Tag getters
+    ///
+    pub fun getTag(id: UInt64): EnglishPremierLeague.Tag? {
+        return EnglishPremierLeague.tagByID[id]
+    }
+
+    /// A top level Play with a unique ID
+    ///
+    pub struct Play {
+        pub let id: UInt64
+        pub let metadata: {String: String}
+        pub var tagIds: [UInt64]
+
+        init (id: UInt64, metadata: {String: String}, tagIds: [UInt64]) {
+            if let play = EnglishPremierLeague.playByID[id] {
+                self.id = play.id
+                self.metadata = play.metadata
+                self.tagIds = play.tagIds
+            } else {
+                self.id = id
+                self.metadata = metadata
+                self.tagIds = tagIds
+            }
+        }
+    }
+
+    /// Play getters
+    ///
+    pub fun getPlay(id: UInt64): EnglishPremierLeague.Play? {
+        return EnglishPremierLeague.playByID[id]
+    }
+
+    /// A public struct to access Edition data
+    ///
+    pub struct Edition {
+        pub let id: UInt64
+        pub let seriesID: UInt64
+        pub let setID: UInt64
+        pub let playID: UInt64
+        pub var maxMintSize: UInt64?
+        pub let tier: String
+        pub var numMinted: UInt64
+
+        init (id: UInt64, seriesID: UInt64, setID: UInt64, playID: UInt64, maxMintSize: UInt64?, tier: String) {
+            if let edition = EnglishPremierLeague.editionByID[id] {
+                self.id = edition.id
+                self.seriesID = edition.seriesID
+                self.setID = edition.setID
+                self.playID = edition.playID
+                self.maxMintSize = edition.maxMintSize
+                self.tier = edition.tier
+                self.numMinted = edition.numMinted
+            } else {
+                self.id = id
+                self.seriesID = seriesID
+                self.setID = setID
+                self.playID = playID
+                self.maxMintSize = maxMintSize
```

Comment:
line 227 should be removed?

Reply:
Yep, it is unnecessary. Will do a fast follow PR to remove. Do not want to reset approvals with a push. Thanks Jude.